scripts/install/setup.sh

```diff
@@ -1,21 +1,55 @@
 #!/usr/bin/env bash
 
-bold() {
-  echo ". $(tput bold)" "$*" "$(tput sgr0)";
-}
-
 err() {
   echo "$*" >&2;
 }
 
-~/spinnaker-for-gcp/scripts/manage/check_duplicate_dirs.sh || exit 1
-~/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
+[ -n "$PARENT_DIR" ] || PARENT_DIR="$HOME"
+
+source $PARENT_DIR/spinnaker-for-gcp/scripts/manage/service_utils.sh
+
+check_for_required_binaries
+
+PARENT_DIR=$PARENT_DIR $PARENT_DIR/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
 
-PROPERTIES_FILE="$HOME/spinnaker-for-gcp/scripts/install/properties"
+[ -z "$PROPERTIES_FILE" ] && PROPERTIES_FILE="$PARENT_DIR/spinnaker-for-gcp/scripts/install/properties"
 
 source "$PROPERTIES_FILE"
 
-~/spinnaker-for-gcp/scripts/manage/check_project_mismatch.sh
+check_for_shared_vpc $CI
+
+PARENT_DIR=$PARENT_DIR PROPERTIES_FILE=$PROPERTIES_FILE $PARENT_DIR/spinnaker-for-gcp/scripts/manage/check_project_mismatch.sh
+
+if [ "$CI" = true ]; then
+  OPERATOR_SA_EMAIL=$(gcloud config list account --format "value(core.account)")
+
+  SETUP_REQUIRED_ROLES=(cloudfunctions.developer compute.networkViewer container.admin iam.serviceAccountCreator iam.serviceAccountUser pubsub.editor redis.admin serviceusage.serviceUsageAdmin source.admin storage.admin)
+  SETUP_EXISTING_ROLES=$(gcloud projects get-iam-policy --filter bindings.members:$OPERATOR_SA_EMAIL $PROJECT_ID \
+    --flatten bindings[].members --format="value(bindings.role)")
+
+  if [ -z "$SETUP_EXISTING_ROLES" ]; then
+    bold "Unable to verify that the service account \"$OPERATOR_SA_EMAIL\" has the required IAM roles."
+    bold "\"$OPERATOR_SA_EMAIL\" requires the IAM role \"Project IAM Admin\" to proceed."
+    exit 1
+  fi
+
+  MISSING_ROLES=""
+  for r in "${SETUP_REQUIRED_ROLES[@]}"; do
+    if [ -z "$(echo $SETUP_EXISTING_ROLES | grep $r)" ]; then
+      if [ -z $MISSING_ROLES ]; then
```

Comment:
```suggestion
      if [ -z "$MISSING_ROLES" ]; then
```

Reply:
Otherwise, we get errors like these:

```
/home/duftler/spinnaker-for-gcp/scripts/install/setup.sh: line 40: [: cloudfunctions.developer,: binary operat
or expected
MISSING_ROLES=cloudfunctions.developer, compute.networkViewer, container.admin
/home/duftler/spinnaker-for-gcp/scripts/install/setup.sh: line 40: [: too many arguments
```

Reply:
:+1:  Thank you